sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -362,6 +362,19 @@ class QueryCompilationErrorsSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SECOND_FUNCTION_ARGUMENT_NOT_INTEGER: " +
+    "the second argument of 'date_add' function needs to be an integer") {
+    withSQLConf(SQLConf.ANSI_ENABLED.key -> "false") {
+      val e = intercept[AnalysisException] {
+        sql("select date_add('1982-08-15', 'x')").collect()
+      }
+      assert(e.getErrorClass === "SECOND_FUNCTION_ARGUMENT_NOT_INTEGER")
+      assert(e.getSqlState === "22023")
+      assert(e.getMessage ===
+        "The second argument of 'date_add' function needs to be an integer.")
```

Comment:
Could you use the `checkErrorClass` method added by https://github.com/apache/spark/pull/36287.